libcloud/common/openstack.py

```diff
@@ -192,12 +193,11 @@ def request(self, action, params=None, data='', headers=None,
         if method.upper() in ['POST', 'PUT'] and default_content_type:
             headers = {'Content-Type': default_content_type}
 
-        return super(OpenStackBaseConnection, self).request(action=action,
-                                                            params=params,
-                                                            data=data,
-                                                            method=method,
-                                                            headers=headers,
-                                                            raw=raw)
+        return retry(self.timeout, self.retry_delay,
```

Comment:
I think this is unnecessary.  The ```super(OpenStackBaseConnection, self).request()``` method already does a ```retry()``` internally.  This will just make exponential retries.  Can you double check that?